core/src/main/scala/org/broadinstitute/dsde/rawls/expressions/ExpressionParser.scala

```diff
@@ -1,29 +1,28 @@
 package org.broadinstitute.dsde.rawls.expressions
 
-import org.broadinstitute.dsde.rawls.model.{AttributeString, MethodConfiguration, ValidatedMethodConfiguration}
+import org.broadinstitute.dsde.rawls.model.{AttributeString, ParsedMCExpressions}
 
 import scala.util.{Failure, Success, Try}
-
 import spray.json._
 
 // a thin abstraction layer over SlickExpressionParser
 
 object ExpressionParser {
-  def parseMCExpressions(methodConfiguration: MethodConfiguration, parser: SlickExpressionParser): ValidatedMethodConfiguration = {
+  def parseMCExpressions(inputs: Map[String, AttributeString], outputs: Map[String, AttributeString], parser: SlickExpressionParser): ParsedMCExpressions = {
     def parseAndPartition(m: Map[String, AttributeString], parseFunc:String => Try[Unit] ) = {
```

Comment:
I feel like this could be achieved with cats' `Validated`. I might post some code showing how that would look, feel free to use or ignore.

Reply:
A rough sketch - was going to PR it but it touches a lot of other code. :) My idea was to model expression validation as a `ValidatedNel[MCValidationError, ParsedMCExpressions]`, where the RHS contains successful inputs/outputs, and the LHS is a list of validation errors.

Again, you don't need to do this for the PR, this was mostly for fun. :)

```
package org.broadinstitute.dsde.rawls.expressions

import cats.data.{Validated, ValidatedNel}
import cats.data.Validated.{Invalid, Valid}
import cats.implicits._
import org.broadinstitute.dsde.rawls.model.{AttributeString, InvalidInput, InvalidOutput, MCValidationError, ParsedMCExpressions}
import spray.json._

import scala.util.Try

// a thin abstraction layer over SlickExpressionParser

object ExpressionParser {

  // Would go in WorkspaceModel
  sealed trait MCValidationError {
    def key: String
    def message: String
  }
  case class InvalidInput(key: String, message: String) extends MCValidationError
  case class InvalidOutput(key: String, message: String) extends MCValidationError

  // Returns a ValidatedNel[MCValidationError, ParsedMCExpressions]
  // The right-hand-side contains only the successes, the left is a list of validation errors
  def parseMCExpressions(inputs: Map[String, AttributeString], outputs: Map[String, AttributeString], parser: SlickExpressionParser): ValidatedNel[MCValidationError, ParsedMCExpressions] = {
    val validatedInputs = parse(inputs, parseAttributeExpr(parser), InvalidInput.apply)
    val validatedOutputs = parse(outputs, parseOutputExpr(parser), InvalidOutput.apply)
    (validatedInputs |@| validatedOutputs).map((validInputs, validOutputs) => ParsedMCExpressions(validInputs, validOutputs))
  }

  private def parse(m: Map[String, AttributeString],
                    parseFunc: String => Validated[Throwable, Unit],
                    errorFunc: (String, String) => MCValidationError): ValidatedNel[MCValidationError, List[String]] = {
    // traverseU is the same oepration as traverse but works around some compiler bugs.
    // Might be removed in Scala 2.12/cats 1.0.
    m.toList.traverseU { case (key, attr) =>
      parseFunc(attr.value) match {
        case Valid(_) =>   Valid(key).toValidatedNel
        case Invalid(e) => Invalid(errorFunc(key, e.getMessage)).toValidatedNel
      }
    }
  }

  private def parseAttributeExpr(parser: SlickExpressionParser)(expression: String): Validated[Throwable, Unit] = {
    Validated.fromTry {
      Try(expression.parseJson).recoverWith { case _ =>
        parser.parseAttributeExpr(expression)
      }.void
    }
  }

  private def parseOutputExpr(parser: SlickExpressionParser)(expression: String): Validated[Throwable, Unit] = {
    Validated.fromTry(parser.parseOutputExpr(expression).void)
  }

}
```